pylint/extensions/redefined_loop_name.py

```diff
@@ -0,0 +1,85 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Optional checker to warn when loop variables are overwritten in the loop's body."""
+
+from astroid import nodes
+
+from pylint import checkers, interfaces
+from pylint.checkers import utils
+from pylint.checkers.variables import in_for_else_branch
+from pylint.interfaces import HIGH
+from pylint.lint import PyLinter
+from pylint.utils import utils as pylint_utils
+
+
+class RedefinedLoopName(checkers.BaseChecker):
+
+    __implements__ = interfaces.IAstroidChecker
```

Comment:
This is no longer necessary and should in fact raise a `DeprecationWarning`.

```suggestion
```

Reply:
Thanks for the heads up. BTW the deprecation warning is being silenced here:
https://github.com/jacobtylerwalls/pylint/blob/0e35511ea2f55da8eaa3272468f320195b5f8006/pylint/checkers/base_checker.py#L189

Reply:
nvm, I was confused, the warning did appear here: https://github.com/PyCQA/pylint/runs/6203141616?check_suite_focus=true